### PR TITLE
[Kubernetes Integration] Updating kubernetes volume dashboard

### PR DESCRIPTION
**Pull request by gizas**
- Enhancement

## Proposed commit message

- WHAT: Updating the Kubernetes Volume Dashboard to include Volume information per Pod
- WHY: Adding  more meaningful information because until now we were providing only general information for the volumes

**Notes:**

- Volume % : Defined as Maximum value
- A filter datastream.dataset=kubernetes.volume added in the visualisation
 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone this repo
- cd to `integrations/build/packages/kubernetes`
- Build latest integaration with `elastic-package build`
- Create a new local elastic stack with `elastic-package stack up -d -v --version=8.15.0-SNAPSHOT`
- Install K8s integration package (latest that this manifest defines is 1.61.2) 
- Navigate to Kubernete Volume Dashboard

## Related issues

- Closes #https://github.com/elastic/beats/issues/39523

## Screenshots

Screenshot 2024-06-26 at 12 13 22 PM

Screenshot 2024-06-26 at 12 13 44 PM

### CODE REVIEW

File: packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json [hunk 1]
```diff
@@ -33,465 +33,694 @@
...
+                                        "sampling": 1
+                                    }
+                                }
                             },
-                            "orderDirection": "desc",
-                            "otherBucket": true,
-                            "parentFormat": {
-                              "id": "terms"
+                            "indexpattern": {
+                                "layers": {}
```
Comment (dej611) [inline packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json:215]: This is not supposed to be here. How did you generate the panel?

File: packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json [hunk 2]
```diff
@@ -33,465 +33,694 @@
...
-                          "label": "Fs Available bytes",
-                          "operationType": "average",
-                          "params": {
-                            "format": {
-                              "id": "bytes",
-                              "params": {
-                                "decimals": 1
-                              }
+                            "textBased": {
+                                "layers": {}
```
Comment (dej611) [inline packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json:218]: This is not supposed to stay here. How did you generate the panel?

Comment (gizas) [reply packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json:218]: I run `elastic-package export dashboards --allow-snapshot` and then I took the part that I changed and included in the specific kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json.

But all the old dashboards have it for eg. https://github.com/elastic/integrations/blob/main/packages/kubernetes/kibana/dashboard/kubernetes-0a672d50-bcb1-11ec-b64f-7dd6e8e82013.json#L162 ?

File: packages/kubernetes/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: 1.61.2
+  changes:
+    - description: Adding Volume Usage per pod in Kubernetes Volume Dashboard
+      type: bugfix
```
Comment (tetianakravchenko) [inline packages/kubernetes/changelog.yml:5]: shouldn't it be an `enhancement`?

File: packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json (hunk 2 above)
Comment (tetianakravchenko) [reply packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json:218]: maybe you used not the latest version of `elastic-package` ?

File: packages/kubernetes/changelog.yml [hunk 4]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: 1.61.2
```
Comment (tetianakravchenko) [inline packages/kubernetes/changelog.yml:2]: please merge in the main branch

Comment (gizas) [reply packages/kubernetes/changelog.yml]: done

**Comment by tetianakravchenko** [comment]
@gizas there is smth wrong:
before this PR - there were 2 vizualisations:
Screenshot 2024-07-05 at 14 06 37

after - 3 with wrong formating, one of them is a duplication:
Screenshot 2024-07-05 at 14 08 20

**Comment by tetianakravchenko** [comment]
after clarification:
1. please add a title for the new vizualisation - like `Volume usage [Metrics Kubernetes]` (similar to other names)
2. clarify if pod name should be added to the `Filesystem Inodes Informations [Metrics Kubernetes]` and `Filesystem Informations [Metrics Kubernetes]` vizualisations

File: packages/kubernetes/changelog.yml [hunk 5]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: 1.62.1
```
Comment (tetianakravchenko) [inline packages/kubernetes/changelog.yml:2]:
```suggestion
- version: 1.63.0
```
for enhancement

Comment (gizas) [reply packages/kubernetes/changelog.yml]: thanks done

**Comment by gizas** [comment]
@eedugon lets clarify above questions: https://github.com/elastic/integrations/pull/10267#issuecomment-2210794216

I have added title `Volume Usage per Pod[Metrics Kubernetes]`

File: packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json (hunk 2 above)
Comment (gizas) [reply packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json]: `elastic-package export dashboards --allow-snapshot
elastic-package version
2024/07/05 16:12:13  INFO New version is available - v0.101.2. Download from: https://github.com/elastic/elastic-package/releases/tag/v0.101.2
elastic-package v0.100.0 version-hash 2632478 (build time: 2024-05-06T13:39:39+03:00)
`

The 0.100.0 is not old, should I upgrade? 

Still exists here with another test: https://github.com/elastic/integrations/pull/10267/files#diff-74341ac7e6879aa8722c1fa981085697141839157d8325f2c4ff9a8536d74485R209-R210

@dej611  the question is that should I also extract the other visualisations?
I was using 8.15.0-SNAPSHOT. Should I use 8.14.1?

Comment (dej611) [reply packages/kubernetes/kibana/dashboard/kubernetes-3912d9a0-bcb2-11ec-b64f-7dd6e8e82013.json]: I suspect that is an issue we should solve on the kibana visualization side.
So for now I'll approve as it does not apply here. 👍

**Review by dej611 — APPROVED** [review]

**Review by tetianakravchenko — APPROVED** [review]
LGTM!
Please clarify https://github.com/elastic/integrations/pull/10267#issuecomment-2210850588 before merging.

**Comment by gizas** [comment]
Synced with @eedugon offline. There was a comment to change the filter of visualisation to metricset.name but we decide not. Reason was to keep it aligned as per other dashboards that have data_stream.dataset used.

This dashboard might need another update when the type of volume will be intorduced

**Pull request merged**